### PR TITLE
Fix bug in getting pkcs7 certs from keys

### PR DESCRIPTION
**Pull request by lumjjb**
Fixes https://github.com/containers/skopeo/issues/1320

Signed-off-by: Brandon Lum <lumjjb@gmail.com>

### CODE REVIEW

**Comment by lumjjb** [comment]
@pravinrajr9 this was changed from the keyprovider commit, can you verify that this fix doesn't break anything on keyprovider side?

**Comment by stefanberger** [comment]
Looks like this broke it? https://github.com/containers/ocicrypt/commit/cabb320230fabd790dd24c5ff66ce9bff5f6f87a#diff-6190ad5ff0b174f39a655dfa268152d635b099a47d1f30b5c1cfff95c87cb8f8L246

Fix lgtm considering the history.

**Comment by lumjjb** [comment]
Yea that seems to be the case.

@stefanberger can you do a github approval as well, i have enabled the protection checks to explicitly check for that now, so will need the green tick. thanks!

File: helpers/parse_helpers.go [hunk 1]
```diff
@@ -207,14 +207,13 @@ func CreateDecryptCryptoConfig(keys []string, decRecipients []string) (encconfig
 		return encconfig.CryptoConfig{}, err
 	}
 
-	if len(x509s) > 0 {
```
Comment (pravinrajr9) [inline helpers/parse_helpers.go:210]: oh my bad, I thought x509 certs are mandatorily  required for protocols other than keyprovider.  But the fix could fail for key provider as we wont be passing any files, and this function would through up "no such file or directory" error.
Could we add a check in this function `processx509Certs `  read file only if it exists, if its not a file doesnt exists then return nil, nil  

`func processx509Certs(keys []string) ([][]byte, error) {
	var x509s [][]byte
	for _, key := range keys {
                if _, err := os.Stat(strings.Split(key, ":")[0]); err == nil {
		tmp, err := ioutil.ReadFile(strings.Split(key, ":")[0])
		if err != nil {
			return nil, errors.Wrap(err, "Unable to read file")
		}
            
		if !encutils.IsCertificate(tmp) {
			continue
		}
		x509s = append(x509s, tmp)
            }
	}
	return x509s, nil
}`

Comment (lumjjb) [reply helpers/parse_helpers.go:210]: I see yea we can do something like that.

Comment (lumjjb) [reply helpers/parse_helpers.go:210]: I just pushed a commit - can you verify on your side that this will work with keyprovider?

**Review by pravinrajr9 — APPROVED** [review]
LGTM

File: helpers/parse_helpers.go (hunk 1 above)
Comment (pravinrajr9) [reply helpers/parse_helpers.go:210]: Oh I already approved. Fix looks good to me, you can go ahead and merge. I will validate tomorrow.

**Comment by stefanberger** [comment]
LGTM

**Review by stefanberger — APPROVED** [review]
LGTM

**Pull request merged**